src/Nethermind/Nethermind.Synchronization/ParallelSync/MultiSyncModeSelector.cs

```diff
@@ -463,8 +463,8 @@ private bool ShouldBeInFastBodiesMode(Snapshot best)
         {
             bool fastBodiesNotFinished = !FastBlocksBodiesFinished;
             bool fastHeadersFinished = FastBlocksHeadersFinished;
-            bool notInStateSync = !best.IsInStateSync;
-            bool stateSyncFinished = best.State > 0;
+            bool notInStateSync = !best.IsInStateSync || _syncConfig.AllowBodiesSyncBeforeStateFinish;
+            bool stateSyncFinished = best.State > 0 || _syncConfig.AllowBodiesSyncBeforeStateFinish;
```

Comment:
Recommend making it a different variable, like:
```
bool canStartDuringStateSync = _syncConfig.AllowBodiesSyncBeforeStateFinish;

bool result = fastBodiesNotFinished && fastHeadersFinished && (notInStateSync && stateSyncFinished) || canStartDuringStateSync);
```

Otherwise, the name of the var is wrong.